packages/core/src/interpreter.ts

```diff
@@ -409,16 +406,33 @@ export class Interpreter<
             State.from(initialState, this.machine.context)
           );
 
-    this.scheduler.initialize(() => {
-      this.update(resolvedState, initEvent as SCXML.Event<TEvent>);
+    this._state = resolvedState;
+
+    this.update(resolvedState);
+
+    if (this.options.devTools) {
+      this.attachDevTools();
+    }
+
+    this.flush();
 
-      if (this.options.devTools) {
-        this.attachDevTools();
-      }
-    });
     return this;
   }
 
+  private flush() {
+    this.mailboxStatus = 'processing';
+    while (this.mailbox.length) {
+      const event = this.mailbox.shift()!;
+
+      this.forward(event);
+
+      const nextState = this.nextState(event);
+
+      this.update(nextState);
```

Comment:
I'm a little bit worried about potential errors coming from those calls, this goes a little bit into the overall error handling story and we probably should not try to handle this anyhow as part of this PR. This is a crucial part of the implementation though so I would add a TODO comment to revisit this once we decide what to do about errors

Reply:
```suggestion
      // TODO: handle errors
      this.forward(event);

      const nextState = this.nextState(event);

      this.update(nextState);
```